sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/transforms/RunInference.java

```diff
@@ -0,0 +1,102 @@
+public class RunInference extends PTransform<PCollection<?>, PCollection<Row>> {
+  private final String modelLoader;
+  private final Schema schema;
+  private final Map<String, Object> kwargs;
+  private final String expansionService;
+
+  /**
+   * Instantiates a multi-language wrapper for a Python RunInference with a given model loader.
+   *
+   * @param modelLoader A Python lambda function for a model loader class object.
+   * @param exampleType A schema field type for the example column in output rows.
+   * @param inferenceType A schema field type for the inference column in output rows.
+   * @return A {@link RunInference} for the given model loader.
+   */
+  public static RunInference of(
+      String modelLoader, Schema.FieldType exampleType, Schema.FieldType inferenceType) {
+    Schema schema =
+        Schema.of(
+            Schema.Field.of("example", exampleType), Schema.Field.of("inference", inferenceType));
+    return new RunInference(modelLoader, schema, ImmutableMap.of(), "");
+  }
+
+  /**
+   * Instantiates a multi-language wrapper for a Python RunInference with a given model loader.
+   *
+   * @param modelLoader A Python lambda function for a model loader class object.
+   * @param schema A schema for output rows.
+   * @return A {@link RunInference} for the given model loader.
+   */
+  public static RunInference of(String modelLoader, Schema schema) {
+    return new RunInference(modelLoader, schema, ImmutableMap.of(), "");
+  }
+
+  /**
+   * Sets keyword arguments for RunInference constructor.
+   *
+   * @return A {@link RunInference} with keyword arguments.
+   */
+  public RunInference withKwarg(String key, Object arg) {
+    ImmutableMap.Builder<String, Object> builder =
+        ImmutableMap.<String, Object>builder().putAll(kwargs).put(key, arg);
+    return new RunInference(modelLoader, schema, builder.build(), expansionService);
+  }
+
+  /**
+   * Sets an expansion service endpoint for RunInference.
+   *
+   * @param expansionService A URL for a Python expansion service.
+   * @return A {@link RunInference} for the given expansion service endpoint.
+   */
+  public RunInference withExpansionService(String expansionService) {
+    return new RunInference(modelLoader, schema, kwargs, expansionService);
+  }
+
+  private RunInference(
+      String modelLoader, Schema schema, Map<String, Object> kwargs, String expansionService) {
+    this.modelLoader = modelLoader;
+    this.schema = schema;
+    this.kwargs = kwargs;
+    this.expansionService = expansionService;
+  }
+
+  @Override
+  public PCollection<Row> expand(PCollection<?> input) {
+    return input.apply(
+        PythonExternalTransform.<PCollection<?>, PCollection<Row>>from(
+                "apache_beam.ml.inference.base.RunInference.create", expansionService)
+            .withKwarg("model_handler_provider", PythonCallableSource.of(modelLoader))
+            .withKwargs(kwargs)
+            .withOutputCoder(RowCoder.of(schema)));
```

Comment:
I'm surprised that PythonExternalTransform has `withOutputCoder` and `withOutputCoders` methods. Shouldn't the Python SDK be responsible for inferring the coder/schema for the PTransform's output?

Reply:
Yes, ideally the Python SDK should be all responsible for inferring output coders but in reality there are many instances that the type inference doesn’t work as expected. Users may want to use `withOutputCoder` method for giving a hint to the Python SDK (it’s optional and not guaranteed to be satisfied).

Reply:
I see. Could we then record the instances where the Python SDK is falling short? It would be nice if there was a path toward deprecating and removing this workaround.

Reply:
> for cross language this information must be provided explicitly

Sure, but shouldn't it be the responsibility of the expanding SDK to figure this out? Not the calling SDK (and potentially the user)

Reply:
The SDK (in this case) figures out it's "Python Object." This essentially says "give it back to me as a a string [or fail]."